tests/reference/asr-test_assign_type_mismatch-f9cd94e.stderr

```diff
@@ -1,4 +1,4 @@
-semantic error: Type mismatch in assignment, the types must be compatible
+semantic error: Type mismatch for an assignment. In LPython, the types must be compatible.
```

Comment:
I think the types must be compatible not just in LPython, but in Python in general. If you try "mypy", what does it say for this case?

Reply:
I actually like the previous message better I think. What we could do is to add multiple spans here with labels and improve the printing of the error message, then we can add more information, such as `In Python the types must be compatible" or something like that.

Reply:
_mypy_ won't give an error.
```console
$ cat a.py
def f():
  a: int
  a = "string"
  print(a)

f()
```
```console
$ python a.py
string
```
```console
$ mypy a.py
Success: no issues found in 1 source file
```

Reply:
That seems like a bug in mypy, isn't it? The type annotation for `a` is `int`, so how can it be a "string"? Unless mypy allows changing the type? Ok, try this:
```
a: int = "string"
```

Reply:
This isn't a bug :) Changing the type is actually common, so we can use` --strict`:
```console
% mypy a.py --strict
a.py:1: error: Function is missing a return type annotation
a.py:1: note: Use "-> None" if function does not return a value
a.py:2: error: Incompatible types in assignment (expression has type "str", variable has type "int")
a.py:5: error: Call to untyped function "f" in typed context
Found 3 errors in 1 file (checked 1 source file)
```

Reply:
I don't know what the default invocation of `mypy` without the `--strict` flag does, but MyPy indeed does not by default allow redefinitions of a type as documented here: https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-allow-redefinition, and even if you enable redefinitions with the `--allow-redefinition` flag, it will still not work in the example above, as documented at that page. You have to use the variable first.

All in all, the types must be compatible not just in LPython, but in CPython also, as enforced by MyPy.